src/dodal/devices/i22/fswitch.py

```diff
@@ -0,0 +1,43 @@
+import asyncio
+import time
+from enum import Enum
+from typing import Dict
+
+from bluesky.protocols import Reading
+from ophyd_async.core import StandardReadable
+from ophyd_async.core.device import DeviceVector
+from ophyd_async.epics.signal import epics_signal_r
+
+
+class FilterState(str, Enum):
+    IN = "IN"
+    OUT = "OUT"
+
+
+class FSwitch(StandardReadable):
+    """
+    Device for i22's fswitch. This currently only implements the minimum
+    functionality for retrieving the number of lenses inserted.
+
+    Eventually this should be combined with the transfocator device in the i04
+    module but is currently incompatible as the Epics interfaces are different.
+    """
+
+    NUM_FILTERS = 128
+
+    def __init__(self, prefix: str, name: str = "") -> None:
+        self.filters = DeviceVector(
+            {
+                i: epics_signal_r(FilterState, f"{prefix}FILTER-{i:03}:STATUS_RBV")
+                for i in range(FSwitch.NUM_FILTERS)
+            }
+        )
+        super().__init__(name)
+
+    async def read(self) -> Dict[str, Reading]:
+        await super().read()
```

Comment:
Should do something with the result of this, for when we make a standard way of reading children devices

Reply:
```python
default_reading = await super().read()
return {"num_filters": Reading(value=num_in, timestamp=time.time()), **default_reading}
```